### PR TITLE
[ws-daemon] Pass error by value instead of reference

### PR DESCRIPTION
**Pull request by princerachit**
fixes https://github.com/gitpod-io/gitpod/issues/5349

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **aledbf** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @aledbf` in a comment when ready.

Associated issue: *#5349*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/ws-daemon/pkg/content/archive.go
Comment (svenefftinge) [inline components/ws-daemon/pkg/content/archive.go:88]: nice!

File: components/ws-daemon/pkg/content/archive.go
Comment (svenefftinge) [inline components/ws-daemon/pkg/content/archive.go:89]: shouldn't we first close the file handle and the delete the file?

Comment (princerachit) [reply components/ws-daemon/pkg/content/archive.go:89]: The defer statements follow LIFO model. The first defer is executed at the last whilst the last defer statement is executed first.

Some examples here: https://www.geeksforgeeks.org/defer-keyword-in-golang/

**Comment by princerachit** [comment]
I tested this in preview env and it works as expected. I request the reviewers to test out this PR .

**Review by MrSimonEmms — APPROVED** [review]
Confirmed this fixes the issue when saving changes 🥳

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: e4142c096ba63c675e862ec26375bcbaa82a727f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5358#pullrequestreview-737421988" title="Approved">MrSimonEmms</a>*

Associated issue: *#5349*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/ws-daemon/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-daemon/OWNERS)~~ [MrSimonEmms]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
